### PR TITLE
Update CrudTrait.php

### PR DESCRIPTION
**Pull request by DelightedD0D**
Updated `getPossibleEnumValues` to use `$instance->getConnectionName()` so that enum values are correctly queried when the Model uses a non-default database connection.

See this issue [#649 ](https://github.com/Laravel-Backpack/CRUD/issues/649)

### CODE REVIEW

File: src/CrudTrait.php [hunk 1]
```diff
@@ -17,7 +17,8 @@
     public static function getPossibleEnumValues($field_name)
     {
         $instance = new static(); // create an instance of the model to be able to get the table name
-        $type = DB::select(DB::raw('SHOW COLUMNS FROM `'.Config::get('database.connections.'.env('DB_CONNECTION').'.prefix').$instance->getTable().'` WHERE Field = "'.$field_name.'"'))[0]->Type;
+        $connectionName = $instance->getConnectionName();
+        $type = DB::connection($connectionName)->select(DB::raw('SHOW COLUMNS FROM `'.Config::get('database.connections.'.$connectionName.'.prefix').$instance->getTable().'` WHERE Field = "'.$field_name.'"'))[0]->Type;
```
Comment (lloy0076) [inline src/CrudTrait.php]: It's possible we should do some escaping in this raw query (but maybe I'm being paranoid?

Comment (DelightedD0D) [inline src/CrudTrait.php]: I would consider any values returned by `Config::get()` and `env()` to be trusted content, no?

Comment (DelightedD0D) [inline src/CrudTrait.php]: Unless an application allows user input to be set to config variables at runtime, in which case I'd argue that it's abusing config variables.

Comment (lloy0076) [inline src/CrudTrait.php]: https://laravel.com/docs/5.4/helpers#method-env -- gets the value of an environment variable which technically means that some malicious entity could hijack a non-cached configuration (I think a real environment variable wins out over even .env variables).

I suppose both of these cases shouldn't occur _too often_ but it is possible.

**Comment by Sobak** [comment]
Please note this change has an additional benefit of removing `env()` usage outside the `config/` dir which is advised by the Laravel documentation and in fact required to work when `php artisan config:cache` is used :)

**Comment by tabacitu** [comment]
Thanks a lot @DelightedD0D ! Merged. Will be available publicly later today, together with a few more bug fixes.

Cheers!

**Pull request closed without merging**